openfoodfacts-implementation/data-generation.ipynb

```diff
@@ -0,0 +1,299 @@
+    "with open('./data/calculated_food_info.json', 'w') as f:\n",
+    "    json.dump(food_info, f, indent=4)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The previous operation may have taken a long time. We may have logged out once the calculations.\n",
+    "# Run this cell if you want to load the JSON with the OpenFoodFacts food information.\n",
+    "food_info = json.load(open('./data/calculated_food_info.json', 'r'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We look for those cases in which the values ​​of 'energy_100g',\n",
+    "# 'sugars_100g', 'fiber_100g', 'fat_100g', 'proteins_100g' and 'carbohydrates_100g'\n",
+    "# be, all, 0. If so, the values ​​that had been obtained from ChatGPT will be taken\n",
+    "# and will be replaced by these.\n",
+    "for recipe_name, recipe_info in food_info.items():\n",
+    "    if recipe_info['nutrition']['energy_100g'] == 0 and recipe_info['nutrition']['sugars_100g'] == 0 and recipe_info['nutrition']['fiber_100g'] == 0 and recipe_info['nutrition']['fat_100g'] == 0 and recipe_info['nutrition']['proteins_100g'] == 0 and recipe_info['nutrition']['carbohydrates_100g'] == 0:\n",
+    "        print(recipe_name, recipe_info['nutrition'])\n",
+    "        recipe_info['nutrition']['sugars_100g'] = recipe_info['sugars']\n",
+    "        recipe_info['nutrition']['fiber_100g'] = recipe_info['fiber']\n",
+    "        recipe_info['nutrition']['fat_100g'] = recipe_info['fat']\n",
+    "        recipe_info['nutrition']['proteins_100g'] = recipe_info['protein']\n",
+    "        recipe_info['nutrition']['carbohydrates_100g'] = recipe_info['carbohydrates']\n",
+    "        # The energy had been obtained from ChatGPT, but can be calculated with the following operation to give more precise information: kcal = 9 x fats + 4 x carbohydrates + 4 x proteins\n",
+    "        recipe_info['nutrition']['energy_100g'] = recipe_info['nutrition']['fat_100g']*9 + recipe_info['nutrition']['carbohydrates_100g']*4 + recipe_info['nutrition']['proteins_100g']*4\n",
+    "        print(recipe_name, recipe_info['nutrition'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We take into account what is generated by ChatGOT to calculate nutritional values\n",
+    "# We do this in each of the dishes and we do it in order to try to have more precise information.\n",
+    "# Energy is not taken into account since ChatGPT has given data that does not fit the aforementioned formula (kcal = 9 x fats + 4 x carbohydrates + 4 x proteins).\n",
+    "# For this reason, its value is recalculated.\n",
+    "for recipe_name, recipe_info in food_info.items():\n",
+    "    recipe_info['sugars'] = (recipe_info['nutrition']['sugars_100g'] + recipe_info['sugars'])/2\n",
```

Comment:
Shouldn't it be 
```python
recipe_info['sugars'] = recipe_info['nutrition']['sugars_100g'] * (recipe_info['portion'] / 100)
```
to calculate the gr of each nutritional value in portion?

So that if a portion of waffles is 150gr, we multiply the sugar in 100gr by (150 / 100 -> 1.5)

Or if you want to consider the previous value, keep the mean but you still have to multiply values to get the portions, right?
```python
recipe_info['sugars'] = (recipe_info['nutrition']['sugars_100g'] + recipe_info['sugars'])/2 * (recipe_info['portion'] / 100)
```

Reply:
Fixed. Now it makes `recipe_info[nutritional_value] = (recipe_info['nutrition'][nutritional_value_100g] + recipe_info[nutritional_value])/2 * (recipe_info['portion'] / 100)` in sugars, fiber, fat, protein and carbohydrates